### PR TITLE
fix: added error handling if there are no instances of currentBatch in db

### PR DESCRIPTION
**Pull request by najamF18**

### CODE REVIEW

File: polls/views.py [hunk 1]
```diff
@@ -45,25 +45,35 @@ def get(self, request):
...
-            return render(request, self.template_name, context)
-        url, task_presentation = present_task_for_user(task)
-        context["url"] = url
-        context["task_presentation"] = task_presentation
+            current_batch_gold = CurrentBatchGold.objects.first()
+            current_batch = (
+                current_batch_gold.current_batch_gold if current_batch_gold else ""
+            )
+
+        context = {}
```
Comment (VivekWangoes) [inline polls/views.py:66]: There is no need to context, because in the below you are already add variable.

File: polls/views.py [hunk 2]
```diff
@@ -45,25 +45,35 @@ def get(self, request):
...
+            if task:
+                url, task_presentation = present_task_for_user(task)
+                context = {
+                    "task": task,
+                    "url": url,
+                    "task_presentation": task_presentation,
+                }
         return render(request, self.template_name, context)
 
     def post(self, request):
```
Comment (VivekWangoes) [inline polls/views.py:79]: There is no comment.

Comment (najamF18) [reply polls/views.py:79]: Can you please elaborate what kind of comment do we need to add? Bt comment do you mean doc strings in relevant functions?

File: polls/views.py [hunk 3]
```diff
@@ -45,25 +45,35 @@ def get(self, request):
...
+            tasks_for_user = current_batch.tasks.exclude(annotation__user=request.user)
+            task = tasks_for_user.first()
+            if task:
+                url, task_presentation = present_task_for_user(task)
+                context = {
+                    "task": task,
+                    "url": url,
+                    "task_presentation": task_presentation,
+                }
         return render(request, self.template_name, context)
```
Comment (DeFidelity) [inline polls/views.py:78]: Great code clean-up, would you mind adding a little unit test to assert that this still works as expected? If your answer is negative kindly let me know. @najamF18

File: polls/views.py (hunk 2 above)
Comment (DeFidelity) [reply polls/views.py:79]: @VivekWangoes I don't see a part that requires comment in here, aside from maybe a little unit test to prove that code does the job which it obviously do.

File: polls/views.py (hunk 3 above)
Comment (najamF18) [reply polls/views.py:78]: Thank you. Sure I will add a unit test for this too

Comment (najamF18) [reply polls/views.py:78]: > Thank you. Sure I will add a unit test for this too

@DeFidelity we can add a test case for this but this will redirect it to login url because we need to authenticate using allauth. We can add the asserts to see if it is being redirected to correct url. Should we go ahead with this? Awaiting your response

Comment (DeFidelity) [reply polls/views.py:78]: Okay, great. 
I think it's good for a merge then

**Review by DeFidelity — APPROVED** [review]

File: polls/views.py (hunk 3 above)
Comment (najamF18) [reply polls/views.py:78]: > Okay, great. I think it's good for a merge then

Great. Do we need to add this test to see if redirection is correct?

Comment (DeFidelity) [reply polls/views.py:78]: @najamF18 absolutely No, we have some more important specs to work on, I was told to create new issues for additional functionalities, I'll appreciate if you can check the log for that test that fails in the other PR, the test passed before maybe something changed by mistake, two previously passed test are now failing and that's what caused the error.
At times all that needs to be done is just repushing the code if nothing changed in the test file from the previous one that passed
